src/app/clinic-dashboard/clinic-dashboard.component.ts

```diff
@@ -1,16 +1,45 @@
 import { Component, OnInit } from '@angular/core';
+
+import { LocationResourceService } from '../openmrs-api/location-resource.service';
+import { ActivatedRoute, Router } from '@angular/router';
 @Component({
   selector: 'app-clinic-dashboard',
   templateUrl: './clinic-dashboard.component.html',
   styleUrls: ['./clinic-dashboard.component.css']
 })
 export class ClinicDashboardComponent implements OnInit {
   locationUuid: string;
-
-  constructor() {
+  objLoaderStatus: boolean;
+  locations = [];
+  constructor(private locationResourceService: LocationResourceService,
+    private route: ActivatedRoute, private router: Router) {
+    this.objLoaderStatus = false;
   }
 
   ngOnInit() {
+    this.locationResourceService.getLocations().subscribe((results: any) => {
+      this.locations = results.map((location) => {
+        return {
+          value: location.uuid,
+          label: location.display
+        };
+      });
+
+      this.route.params.subscribe(params => {
+        setTimeout(() => {
+          this.locationUuid = params['location_uuid'];
```

Comment:
Selected location should also be tracked by the cached service, to make it easy for components to consume the currently selected clinic.